substrate-node/pallets/pallet-tfgrid/src/node.rs

```diff
@@ -0,0 +1,211 @@
+pub const MAX_COUNTRY_NAME_LENGTH: u32 = 56;
+
+// 1: Y
+pub const MIN_CITY_NAME_LENGTH: u32 = 1;
+// 85: Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch
+pub const MAX_CITY_NAME_LENGTH: u32 = 58;
+
+/// A location lat/long (ASCI Characters).
+#[derive(Encode, Decode, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct Location<T: Config>(
+    // Latitude
+    pub(crate) BoundedVec<u8, ConstU32<MAX_LOCATION_LENGTH>>,
+    // Longitude
+    pub(crate) BoundedVec<u8, ConstU32<MAX_LOCATION_LENGTH>>,
+    PhantomData<(T, ConstU32<MAX_LOCATION_LENGTH>)>,
+);
+
+impl<T: Config> TryFrom<(Vec<u8>, Vec<u8>)> for Location<T> {
+    type Error = Error<T>;
+
+    /// Fallible initialization from a provided tuple of byte vector
+    /// (latitude and longitude) if one is below the minimum or exceeds
+    /// the maximum allowed length or can not be converted to float or
+    /// is out of [-90; 90] range (for latitude) or is out of [-180; 180]
+    /// range (for longitude)
+    fn try_from(value: (Vec<u8>, Vec<u8>)) -> Result<Self, Self::Error> {
+        ensure!(
+            value.0.len() >= MIN_LOCATION_LENGTH.saturated_into(),
+            Self::Error::NodeLatitudeInputToShort
+        );
+        let latitute_bv: BoundedVec<u8, ConstU32<MAX_LOCATION_LENGTH>> =
+            BoundedVec::try_from(value.0.clone())
+                .map_err(|_| Self::Error::NodeLatitudeInputToLong)?;
+        ensure!(
+            validate_latitude_input(&latitute_bv),
+            Self::Error::InvalidNodeLatitudeInput
+        );
+
+        ensure!(
+            value.1.len() >= MIN_LOCATION_LENGTH.saturated_into(),
+            Self::Error::NodeLongitudeInputToShort
+        );
+        let longitute_bv: BoundedVec<u8, ConstU32<MAX_LOCATION_LENGTH>> =
+            BoundedVec::try_from(value.1.clone())
+                .map_err(|_| Self::Error::NodeLongitudeInputToLong)?;
+        ensure!(
+            validate_longitude_input(&longitute_bv),
+            Self::Error::InvalidNodeLongitudeInput
+        );
+
+        Ok(Self(latitute_bv, longitute_bv, PhantomData))
+    }
+}
+
+// FIXME: did not find a way to automatically implement this.
+impl<T: Config> PartialEq for Location<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0 && self.1 == other.1
+    }
+}
+
+// FIXME: did not find a way to automatically implement this.
+impl<T: Config> Clone for Location<T> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone(), self.1.clone(), self.2)
+    }
+}
+
+fn validate_latitude_input(input: &[u8]) -> bool {
+    match core::str::from_utf8(input) {
+        Ok(val) => {
+            if let Some(lat) = val.parse::<f32>().ok() {
+                lat >= -90.0 && lat <= 90.0
+            } else {
+                false
+            }
+        }
+        Err(_) => false,
+    }
+}
+
+fn validate_longitude_input(input: &[u8]) -> bool {
+    match core::str::from_utf8(input) {
```

Comment:
not sure if you can use this (since this is std ? )

You can try to compile the node once you fixed build errors to check if it compiles or not

Reply:
see previous comment https://github.com/threefoldtech/tfchain/pull/441#discussion_r967683455